resolver/resolver.cc

```diff
@@ -2082,10 +2080,9 @@ class ResolveTypeMembersAndFieldsWalk {
                 job.asgn->rhs = ast::MK::Send1(loc, ast::MK::Constant(loc, core::Symbols::Magic()),
                                                core::Names::suggestType(), loc.copyWithZeroLength(), move(rhs));
             }
-            return resultType;
         }
-
-        return data->resultType;
+        // Always return the re-derived type, even on the fast path.
+        return resultType;
```

Comment:
This is the only part of t he PR I don't really understand the consequences of: we don't skip work if the symbol already had its type set.  I think this only means that we constantly re-assign `job.asgn->rhs` in some cases, but it seems like these cases should be unusual: static fields are probably not usually defined in multiple files or even multiple times in the same file (?).

(We already had this problem a little bit since `resolveConstantType` got moved out of the `if` several PRs ago, so I think the performance implications are already built-in -- and the unusualness of defining the same static field in multiple places also indicates we usually won't suffer performance problems here, I think?)

Reply:
Yeah i'm not really sure.

The only thing I can think of here is to get really creative with a lot of fast-path tests.

I found one failing test for you, which should give you something to think about for the time being.

If you wouldn't mind trying to write a bunch of weird and wacky test cases after you've looked into that, we can revisit these resolver changes.

Suggestions for crafting test cases would be: think of weird things. Use lots of different kinds of symbols. Use weird features like `T::Enum`, type members, class aliases, type aliases, etc. Try to find errors that are more than just the result type (the error below is a resolver/type_syntax.cc error).